plugins/modules/panos_commit_firewall.py

```diff
@@ -0,0 +1,210 @@
+
+DOCUMENTATION = r'''
+---
+module: panos_commit_firewall
+short_description: Commit the firewall's candidate configuration.
+description:
+    - Module that will commit the candidate configuration of a PAN-OS firewall.
+    - The new configuration will become active immediately.
+author:
+    - Robert Hagen (@stealthllama)
+version_added: "2.10"
+requirements:
+    - pan-os-python
+    - dateutils
+extends_documentation_fragment:
+    - paloaltonetworks.panos.fragments.provider
+options:
+    description:
+        description:
+            - A description of the commit.
+        type: str
+    admins:
+        description:
+            - Commit only the changes made by specified list of administrators.
+        type: list
+    exclude_device_and_network:
+        description:
+            - Exclude network and device configuration changes.
+        type: bool
+        default: False
+    exclude_shared_objects:
+        description:
+            - Exclude shared object configuration changes.
+        type: bool
+        default: False
+    exclude_policy_and_objects:
+        description:
+            - Exclude policy and object configuration changes.
+        type: bool
+        default: False
+    force:
+        description:
+            - Force the commit.
+        type: bool
+        default: False
+    sync:
+        description:
+            - Wait for the commit to complete.
+        type: bool
+        default: True
+'''
+
+EXAMPLES = r'''
+- name: commit candidate configs on firewall
+  panos_commit_firewall:
+    provider: '{{ credentials }}'
+
+- name: commit changes by specified admins on firewall
+  panos_commit_firewall:
+    provider: '{{ credentials }}'
+    admins: ['netops','secops','cloudops']
+    description: 'Saturday change window'
+
+- name: commit only policy and object changes on firewall
+  panos_commit_firewall:
+    provider: '{{ credentials }}'
+    exclude_device_and_network: True
+'''
+
+RETURN = r'''
+jobid:
+  description: The ID of the PAN-OS commit job.
+  type: int
+  returned: always
+  sample: 49152
+starttime:
+  description: The time the commit job started.
+  type: str
+  returned: on success
+  sample: 2020, 9, 6, 17, 56, 20
+endtime:
+  description: The time the commit job ended.
+  type: str
+  returned: on success
+  sample: 2020, 9, 6, 17, 56, 35
+runtime:
+  description: The elapsed time (in seconds) of the commit job.
+  type: int
+  returned: on success
+  sample: 15
+details:
+  description: Commit job completion messages.
+  type: str
+  returned: on success
+  sample: Configuration committed successfully
+warning:
+  description: Commit job warnings.
+  type: list
+  returned: on success
+  sample: ["","","","","","","","","","",""]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.paloaltonetworks.panos.plugins.module_utils.panos import get_connection
+
+try:
+    from panos.firewall import FirewallCommit
+    from dateutils import parser
```

Comment:
No extra dependencies...  :(

I'm wondering if returning the start and end time as a timestamp might ultimately be the most useful thing if you want to give that information to customers.